models/src/main/java/com/foodie/FoodieBaseService.java

```diff
@@ -35,11 +31,25 @@ default Optional<T> findById(Key id) {
         return getRepository().findById(id);
     }
 
+    default T save(T entity) {
```

Comment:
I have not seen anyone use `default` modifiers before. Why do you feel you need them here?

Reply:
Adding a default method, or changing a method from abstract to default, does not break compatibility with pre-existing binaries, but may cause an IncompatibleClassChangeError if a pre-existing binary attempts to invoke the method. This error occurs if the qualifying type, T, is a subtype of two interfaces, I and J, where both I and J declare a default method with the same signature and result, and neither I nor J is a subinterface of the other.